File: Bookshelf.Api/Bookshelf.Api.Domain/Entities/Book.cs
using Bookshelf.Api.Domain.Enums;

namespace Bookshelf.Api.Domain.Entities
{
    public class Book
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Author { get; set; }
        public decimal Price { get; set; }
        public GenreType Genre { get; set; }
        public int Publish_Year { get; set; }
        public string Publisher { get; set; }
        public string Description { get; set; }
        public string Image_Attachment { get; set; }
    }
}

File: Bookshelf.Api/Bookshelf.Web/clientApp/src/app/classes/book.ts

export class Book {
  constructor(
    public id: number,
    public title: string,
    public author: string,
    public price: number,
    public genre: 'Fiction' | 'Mystery' | 'Thriller' | 'Horror' | 'Historical' | 'Romance' | 'Western' | 'Science Fiction' | 'Fantasy' | 'Poetry',
    public publish_Year: number,
    public publisher: string,
    public description: string,
    public image_attachment: string,
  ) { }
}
